sbnobj/Common/PMT/Data/PMTBeamSignal.cxx

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file   sbnobj/Common/PMT/Data/PMTBeamSignal.cxx
+ * @brief  Holds the event-by-event RWM or EW times
+ * @author Anna Heggestuen (aheggest@colostate.edu), adapted from M. Vincenzi in https://github.com/SBNSoftware/icaruscode/pull/751
+ * @date   May 19, 2025
+ * @see    sbnobj/Common/PMT/Data/PMTBeamSignal.h
+ */
+
+//library header
+#include "sbnobj/Common/PMT/Data/PMTBeamSignal.hh"
+
+// framework libraries
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+// C/C++ standard libraries 
+#include <iostream>
+
+// -----------------------------------------------------------------------------
+void sbn::timing::SelectFirstOpHitByTime(const recob::OpHit* const hit, 
+                                        std::map<int, double> &startmap, 
+                                        std::map<int, double> &risemap)
+{
+  const int ch = hit->OpChannel();
+  double ts = hit->StartTime();
+  double tr = hit->RiseTime();
+  // select the first ophit (by time) in each channel
+  if (startmap.find(ch) != startmap.end())
+  {
+    if (ts < startmap[ch])
+      {
+        startmap[ch] = ts;
+        risemap[ch] = ts + tr;
+      }
+  }
+  else
+  {
+    startmap[ch] = ts;
+    risemap[ch] = ts + tr;
+  }  
+}
+// -----------------------------------------------------------------------------
+int sbn::timing::getSideByChannel(const int channel)
+{
+  /*
+  Channels are numbered from east to west, from North (cryo side) to South (beam side)
+  We look in the opposide direction wrt to the beam direction South->North:
+  - Left is the east wall of each cryostat;
+  - Right is the west side of each cryostat;
+  - [ 0:89 ] and [180:269] are on the left,
+    the return value of the function is 0;
+  - [ 90-179 ] and [ 270:359 ] are on the right,
+    the return value of the function is 1;
+  */
+
+  int side = channel / 90; // always round down
+  return side % 2;
+}
+
+// -----------------------------------------------------------------------------
+double sbn::timing::getFlashBunchTime(std::map<int, double> startmap, 
+                                      std::map<int, double> risemap,
+                                      std::vector<sbn::timing::PMTBeamSignal> RWMTimes)
```

Comment:
```suggestion
double sbn::timing::getFlashBunchTime(std::map<int, double> risemap,
                                      std::vector<sbn::timing::PMTBeamSignal> RWMTimes)
```
There is no need for `startmap` here. From the code, it looks like `risemap` already contains the first hit rise time relative to the trigger time for each channel. You can simply loop over that map and use `it->second` instead of `risemap[ch]`.

It's worth noting that the rise time in the OpHit is stored relative to the start time. That's why the helper function `sbn::timing::SelectFirstOpHitByTime` does `ts + tr`. I'm assuming that is being called first.

Reply:
thanks Matteo! i removed the startmap from the getFlashBunchTime function. yes, the `SelectFirstOpHitByTime` function is called before `getFlashBunchTime`.